### PR TITLE
DBZ-438 BufferedBlockingConsumer must be synchronized

### PR DESCRIPTION
**Pull request by jpechane**
Initial copy can be done in multiple threads - https://issues.jboss.org/browse/DBZ-438

### CODE REVIEW

**Comment by jpechane** [comment]
As it is accessef from multpile threads it either needs to be synced or a lock should be present. Given we do not need to combine read/write locks and we do not need smaller granularity I opted for `synchronized`.

File: debezium-core/src/main/java/io/debezium/function/BufferedBlockingConsumer.java [hunk 1]
```diff
@@ -51,13 +52,13 @@ public default void flush() throws InterruptedException {
             private T last;
 
             @Override
-            public void accept(T t) throws InterruptedException {
+            public synchronized void accept(T t) throws InterruptedException {
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/function/BufferedBlockingConsumer.java]: I'm concerned a bit about this, IIUC it'll enforce synchronization of all the producer threads which seems bad. I think we could use an `AtomicReference` instead. I'll push a PR for it and you can let me know what you think.

Comment (gunnarmorling) [reply debezium-core/src/main/java/io/debezium/function/BufferedBlockingConsumer.java]: @jpechane I've pushed an alternative here: https://github.com/debezium/debezium/pull/338. Could you check it out?

**Comment by jpechane** [comment]
Superseded with https://github.com/debezium/debezium/pull/338

**Pull request closed without merging**